File: hashset/contais_duplicate2/solution.js
function containsNearbyDuplicate(nums, k) {
  const set = new Set();

  for (let i = 0; i < nums.length; i++) {
    if (set.has(nums[i])) return true;

    set.add(nums[i]);
    if (set.size > k) {
      set.delete(nums[i - k]);
    }
  }

  return false;
}

// Example usage
const nums = [1, 2, 3, 1, 4, 5];
const k = 3;
const result = containsNearbyDuplicate(nums, k);
console.log(result); // Output: true

File: hashset/contais_duplicate2/solution.py
class Solution:
    def containsNearbyDuplicate(self, nums, k):
        seen = set()

        for i, num in enumerate(nums):
            if num in seen:
                return True
            seen.add(num)

            if len(seen) > k:
                seen.remove(nums[i - k])

        return False


# Example usage
if __name__ == "__main__":
    nums = [1, 2, 3, 1, 4, 5]
    k = 3
    sol = Solution()
    result = sol.containsNearbyDuplicate(nums, k)
    print(result)  # Output: True
